src/metabase/query_processor/middleware/process_userland_query.clj

```diff
@@ -141,6 +154,12 @@
    :result_rows       0
    :start_time_millis (System/currentTimeMillis)})
 
+(defn- query->field-usages
+  [query]
+  (->> (normalize/normalize-preprocessing-middleware query)
```

Comment:
I used normalize instead of  `lib/query` because I realized that `query` now can also be a mlv2 query. Here is the PR that introduced it https://github.com/metabase/metabase/pull/39969

Reply:
Can't `lib/query` also handle MLv2? 

Looking at it's implementation, it seems to be designed to handle it here
https://github.com/metabase/metabase/blob/679fd3506c95e1c3b8e1a5194f315ecf647546c7/src/metabase/lib/query.cljc#L128

Reply:
sadly know, because `lib/type` in the JSON query is a string, not a keyword, `normalize/normalize-preprocessing-middleware` will take care of converting.

Reply:
This is misusing the middleware. It's not a helper function intended to be called like this. The middleware runs in a defined order, see `metabase.query-processor.preprocess/middleware` for the big list.

In particular, you'll see that `normalize/normalize-preprocessing-middleware` is the first step, and `fetch-source-query/resolve-source-cards` is near the top. So you shouldn't have to call either of these - the query will be fully preprocessed already.

Reply:
If I understand correctly, these "round" middlewares are executed before the "preprocessing" middleware. So the query here is not full processed.

Reply:
Should I create a preprocessing middleware for this? not sure if it's the right way to introduce side-affects to QP tho.